### PR TITLE
M3 190/add discovery module

### PR DESCRIPTION
**Pull request by doron-st**
Ready to merge. 
Do recalling by default, and independent discovery if --discovery flag is on

### CODE REVIEW

File: nf/maestro3/genotype_fastq.nf [hunk 1]
```diff
@@ -413,11 +448,11 @@ process mergeVCFs {
     memory '3 GB'
 
     input:
-    set val(id), file(input_vcfs), file(input_vcf_indeces) from variant_recalling_vcf_ch
+    set val(id), val(mode), file(input_vcfs), file(input_vcf_indeces) from variant_recalling_vcf_ch.mix(variant_discovery_vcf_ch)
```
Comment (avigailmo) [inline nf/maestro3/genotype_fastq.nf]: what `variant_discovery_vcf_ch` gets if variantDiscovery didn't run?

File: nf/maestro3/genotype_fastq.nf [hunk 2]
```diff
@@ -388,9 +421,11 @@ process variantRecalling {
     disk agg_large_disk
     memory 1500 * params.scatter_count + ' MB'
     cpus params.scatter_count
+    mode = "recalling"
```
Comment (avigailmo) [inline nf/maestro3/genotype_fastq.nf]: needed?

File: nf/maestro3/genotype_fastq.nf (hunk 1 above)
Comment (avigailmo) [reply nf/maestro3/genotype_fastq.nf]: you expecting the values from one channel to be override by the other?
can you please explain which values we are going to get here  in any use case?

Comment (doron-st) [reply nf/maestro3/genotype_fastq.nf]: If a process wasn't run we get an empty channel from its output.
The mix function, does "union" over two channels.
So MergeVCF always gets all the vcf outputs that were created, regardless of parameters

Comment (avigailmo) [reply nf/maestro3/genotype_fastq.nf]: but what about the `mode`?

File: nf/maestro3/genotype_fastq.nf (hunk 2 above)
Comment (doron-st) [reply nf/maestro3/genotype_fastq.nf]: nope

File: nf/maestro3/genotype_fastq.nf (hunk 1 above)
Comment (doron-st) [reply nf/maestro3/genotype_fastq.nf]: the mode is a string passed as part of the channel.
For recalling channels, the mode will be "recalling".
For discovery channels, the mode will be "discovery"
BTW, Its important to create sets/tuples for the channels to make sure we don't mix outputs of different channels together.

Comment (doron-st) [reply nf/maestro3/genotype_fastq.nf]: In mergeVCF we use mode to detemine the name of the merged output VCF.
We want to distinguish the two possible outputs per sample (discovery/recallin)

**Pull request merged**